### PR TITLE
Injimob 2277 handle expired vc

### PR DESCRIPTION
**Pull request by balachandarg-tw**
## Description

Handling Expired Vcs:

1. Showing yellow icon and Expired Status in Home screen and Detailed Screen as per [Figma Design](https://www.figma.com/design/F6HYVibHTUUWicHMFfBgjM/Inji-Wallet-App?node-id=225-705&node-type=frame)

2. Removed Activation shield from Home screen

3. Removed Vc Activation option from Kebab popup menu for both Home and Details screen

4. Removing Vc Activation section in Details screen.

## Issue ticket number and link

> [INJIMOB-2277](https://mosip.atlassian.net/jira/software/c/projects/INJIMOB/boards/91?selectedIssue=INJIMOB-2277)

[INJIMOB-2277]: https://mosip.atlassian.net/browse/INJIMOB-2277?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: machines/Issuers/IssuersMachine.ts [hunk 1]
```diff
@@ -394,14 +394,18 @@ export const IssuersMachine = model.createMachine(
           src: 'verifyCredential',
           onDone: [
             {
-              actions: ['sendSuccessEndEvent', 'setIsVerified'],
+              actions: ['sendSuccessEndEvent', 'setIsVerified', 'setIsExpired'],
```
Comment (swatigoel) [inline machines/Issuers/IssuersMachine.ts:397]: no need to have separate action.
isExpired Vc part of vc verification, so should have single action which makes VC verified and expired if any.

File: machines/Issuers/IssuersMachine.ts [hunk 2]
```diff
@@ -394,14 +394,18 @@ export const IssuersMachine = model.createMachine(
...
             },
           ],
           onError: [
             {
               cond: 'isVerificationPendingBecauseOfNetworkIssue',
-              actions: ['resetLoadingReason', 'resetIsVerified'],
+              actions: [
+                'resetLoadingReason',
+                'resetIsVerified',
+                'resetIsExpired',
```
Comment (swatigoel) [inline machines/Issuers/IssuersMachine.ts:407]: no need to have separate action

File: machines/Issuers/IssuersActions.ts [hunk 3]
```diff
@@ -43,6 +44,21 @@ export const IssuersActions = (model: any) => {
...
+        });
+      },
+    }),
+    resetIsExpired: assign({
+      vcMetadata: (context: any, event: any) =>
+        new VCMetadata({
+          ...context.vcMetadata,
+          isExpired: false,
+        }),
+    }),
```
Comment (swatigoel) [inline machines/Issuers/IssuersActions.ts:61]: no need to have separate action for isVerifed and isExpired.
Have a action to handle verification result.

**Review by swatigoel — CHANGES_REQUESTED** [review]

File: machines/Issuers/IssuersMachine.ts (hunk 1 above)
Comment (balachandarg-tw) [reply machines/Issuers/IssuersMachine.ts]: Done

File: components/VC/Views/VCCardView.tsx [hunk 4]
```diff
@@ -102,6 +102,7 @@ export const VCCardView: React.FC<VCItemProps> = props => {
       KEBAB_POPUP={controller.KEBAB_POPUP}
       isVerified={props.vcMetadata.isVerified}
       isInitialLaunch={props.isInitialLaunch}
+      isExpired={props.vcMetadata.isExpired}
```
Comment (KiruthikaJeyashankar) [inline components/VC/Views/VCCardView.tsx:105]: Do we need to send `isExpired` prop explicitly given `vcMetadata` is already passed?

File: components/VC/Views/VCCardViewContent.tsx [hunk 5]
```diff
@@ -90,6 +90,7 @@ export const VCCardViewContent: React.FC<VCItemContentProps> = props => {
               <VCVerification
                 wellknown={props.wellknown}
                 isVerified={props.isVerified}
+                isExpired={props.isExpired}
```
Comment (KiruthikaJeyashankar) [inline components/VC/Views/VCCardViewContent.tsx:93]: can we get this `isExpired` field from  `vcMetadata` directly if available?

Comment (balachandarg-tw) [reply components/VC/Views/VCCardViewContent.tsx]: Done

**Review by swatigoel — APPROVED** [review]

**Pull request merged**